plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/expression/JdbcConnectorExpressionRewriterBuilder.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.jdbc.expression;
+
+import com.google.common.collect.ImmutableSet;
+import io.trino.plugin.base.expression.ConnectorExpressionRewriter;
+import io.trino.plugin.base.expression.ConnectorExpressionRule;
+
+import java.util.function.Function;
+
+import static java.util.Objects.requireNonNull;
+
+public class JdbcConnectorExpressionRewriterBuilder
+{
+    public static JdbcConnectorExpressionRewriterBuilder newBuilder()
+    {
+        return new JdbcConnectorExpressionRewriterBuilder();
+    }
+
+    private Function<String, String> identifierQuote;
+    private ImmutableSet.Builder<ConnectorExpressionRule<?, String>> rules = ImmutableSet.builder();
+
+    private JdbcConnectorExpressionRewriterBuilder() {}
+
+    public JdbcConnectorExpressionRewriterBuilder withIdentifierQuote(Function<String, String> identifierQuote)
```

Comment:
And if it's optional an empty function is easy to pass - we can make it part of builder if need arises in future.

Reply:
> Won't identifier quote always be needed? If yes, then I'd keep it in the constructor.

I don't believe in "all required fields go into builder constructor".
It's may be acceptable when there is only one or two required field, but generally this kind of defeats the purpose of the builder -- to have parameters provided individually with clearly-named setter methods, instead of positional parameters.

If something is required, it needs to be required, that's all. Requiring in constructor isn't better than checking requirment in `build()` method. The potential gain -- of developers' efficiency -- doesn't apply here, since we rarely write code using this class. We will write it, say, 10 times and that's it. If i were using this API 10 times _a day_, then indeed making it super clear which things are required could be an overall time saver.

Reply:
My point was that to users of this class it'll be obvious that an identifier quote needs to be passed in. `build` would fail at runtime only.

However I don't feel very strongly about this (yet).

Reply:
again

> I don't believe in "all required fields go into builder constructor".

it leads to unnecessarily asymmetric API, that's harder to read.
It might be  beneficial if you implement it frequently, which won't be the case here.

btw i will look into removing the quote at all from here.

Reply:
> btw i will look into removing the quote at all from here.

https://github.com/trinodb/trino/pull/11464

Reply:
Let's follow up on this - we can merge this PR and then continue on the other one.